### DISCUSSION TITLE
Holding movement threshold

### DISCUSSION THREAD

**Discussion by HakanL** [unoplatform/uno#19112] [Q&A] [answered]
I believe I have asked this before but I can't find the question/answer. Is there a way to change the tolerance/threshold for holding a button to a larger value than what it's current? On my small touchscreen (Skia, RPi) I have a button close to the edge of the screen and I want to detect `Holding` on that. The issue is that because of the physical placement it seems to be very easy to move your fingers a few pixels, so a lot of times I don't get the holding event. Is there a setting to make the holding less sensitive? For my purpose I don't really care about cancelling/aborting a hold, if the user has touched the screen then they want to execute.

**Comment by morning4coffe-dev** [comment, marked as answer]
Hey @HakanL,

Thank you for your question! I don't think we/WinUI have a feature like this directly, but you can work around it using the `OnPointerPressed`, `OnPointerMoved`, and `OnPointerReleased` methods. You can start a timer when the button is pressed and stop it if the pointer is released or moves outside a certain threshold. Let me know if this helps!

**Reply by HakanL** [reply]
Yeah, I've done that, but it's annoying to have to do that, and also OnPointerPressed doesn't seem to work on buttons, it's never triggered.

**Reply by morning4coffe-dev** [reply]
Jup, I see, but as I mentioned afaik even WinUI doesn't have this feature, right? Can you open an issue with an enhancement request, maybe it could be considered as a useful Toolkit feature in the future.

**Reply by HakanL** [reply]
You're probably right that WinUI doesn't have that feature, I didn't look as my target platform is Linux and Skia, so WinUI isn't an option. What about `OnPointerPressed` on buttons, is that a WinUI limitation as well?

**Reply by MartinZikmund** [reply]
The reason is that `Button` class itself handles `PointerPressed` so it never bubbles to your handler. It should be possible to create a custom class derived from `Button` to override it with your own logic. Alternative I thought about could be to keep the size of the button visually the same, but adjust its template so that the margin around it is still "interactive" (e.g. by putting an invisible surface there), so that it still is active even if user moves their finger slightly.

**Reply by HakanL** [reply]
Thanks Martin, I'll try that.
